plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/sampler/ProbabilisticSampler.java

```diff
@@ -0,0 +1,75 @@
+package org.opensearch.telemetry.tracing.sampler;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+import org.opensearch.telemetry.OTelTelemetrySettings;
+
+import java.util.List;
+
+/**
+ * ProbabilisticSampler implements a head-based sampling strategy based on provided settings.
+ */
+public class ProbabilisticSampler implements Sampler {
+    private Sampler defaultSampler;
+    private final OTelTelemetrySettings oTelTelemetrySettings;
+
+    private double tracerHeadSamplerSamplingRatio;
+
+    /**
+     * Constructor
+     *
+     * @param oTelTelemetrySettings Telemetry settings.
+     */
+    public ProbabilisticSampler(OTelTelemetrySettings oTelTelemetrySettings) {
+        if (oTelTelemetrySettings == null) {
+            throw new IllegalArgumentException("oTelTelemetrySettings cannot be null");
+        }
+        this.oTelTelemetrySettings = oTelTelemetrySettings;
+        this.tracerHeadSamplerSamplingRatio = oTelTelemetrySettings.getTracerHeadSamplerSamplingRatio();
+        this.defaultSampler = createProbabilitySampler(tracerHeadSamplerSamplingRatio);
+    }
+
+    private Sampler createProbabilitySampler(double samplingRatio) {
+        return Sampler.traceIdRatioBased(samplingRatio);
+    }
+
+    Sampler getSampler() {
+        double newSamplingRatio = oTelTelemetrySettings.getTracerHeadSamplerSamplingRatio();
```

Comment:
This is not thread safe, his may mess with the values in the multi threaded calls.

Reply:
Updated!